hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/create/CreateDecoder.java

```diff
@@ -122,6 +125,32 @@ public TransactionBody decodeCreateFungibleTokenV3(
         return bodyFor(tokenCreateWrapper);
     }
 
+    /**
+     * Decodes a call to {@link CreateTranslator#CREATE_FUNGIBLE_TOKEN_WITH_METADATA} into a synthetic {@link TransactionBody}.
+     *
+     * @param encoded the encoded call
+     * @param senderId the sender account ID
+     * @param nativeOperations the native operations
+     * @param addressIdConverter the address ID converter
+     * @return the synthetic transaction body
+     */
+    public TransactionBody decodeCreateFungibleTokenWithMetadata(
```

Comment:
this class has a lot of static methods in this class now.  Might it be possible to have fewer by checking the number of tuples in the `HederaToken` tuple (i.e. tuple.size())?

Reply:
I removed the static modifiers as I don't see why we need then in first place.
We have separate methods for metadata calls so that we have completely separate flows (with/without metadata).